### PR TITLE
always use @JacksonXmlElementWrapper and read xml data from referenced schema, updated

### PR DESCRIPTION
**Pull request by martin-mfg**
This PR is based on @jzrebiec's commits in #5371. I mainly rebased those to the current master.

The changes can be tested e.g. with [this input file](https://gist.github.com/martin-mfg/7f427836adb64bea947cea0cc3629aad), and the [java-resttemplate-withXml.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/bin/configs/java-resttemplate-withXml.yaml) config.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.
    @bbdouglas @sreeshas @jfiala @lukoyanov @cbornet @jeff9finger @karismann @Zomzog @lwlee2608

### CODE REVIEW

File: samples/client/petstore/java/resttemplate-withXml/src/main/java/org/openapitools/client/model/AdditionalPropertiesClass.java [hunk 1]
```diff
@@ -144,6 +144,9 @@ public AdditionalPropertiesClass putMapStringItem(String key, String mapStringIt
   @javax.annotation.Nullable
   @JsonProperty(JSON_PROPERTY_MAP_STRING)
   @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)
+  // xmlName=
```
Comment (wing328) [inline samples/client/petstore/java/resttemplate-withXml/src/main/java/org/openapitools/client/model/AdditionalPropertiesClass.java:147]: looks like xmlName is complete empty. Is that expected?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java [hunk 2]
```diff
@@ -3927,6 +3919,32 @@ public CodegenProperty fromProperty(String name, Schema p, boolean required, boo
...
+                property.isXmlAttribute = p.getXml().getAttribute();
+            }
+            if (p.getXml().getWrapped() != null) {
+                property.isXmlWrapped = p.getXml().getWrapped();
+            }
+            property.xmlPrefix = p.getXml().getPrefix();
+            property.xmlName = p.getXml().getName();
+            property.xmlNamespace = p.getXml().getNamespace();
+        }
+
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java:3947]: cc @OpenAPITools/generator-core-team

File: samples/client/petstore/java/resttemplate-withXml/src/main/java/org/openapitools/client/model/AdditionalPropertiesClass.java (hunk 1 above)
Comment (martin-mfg) [reply samples/client/petstore/java/resttemplate-withXml/src/main/java/org/openapitools/client/model/AdditionalPropertiesClass.java]: I added a commit to make this change.

**Comment by wing328** [comment]
CI failure already fixedin the master.

**Pull request merged**